src/scales/ordinal.js

```diff
@@ -60,7 +61,7 @@ export function ScaleOrdinal(key, channels, {
     }
   }
   if (unknown === scaleImplicit) throw new Error("implicit unknown is not supported");
-  return ScaleO(scaleOrdinal().unknown(unknown), channels, {...options, type, domain, range, hint});
+  return ScaleO(scaleOrdinal().unknown(unknown), channels, {...options, key, type, domain, range, hint});
```

Comment:
I would prefer not to mix the `key` with options.

```suggestion
  return ScaleO(key, scaleOrdinal().unknown(unknown), channels, {...options, type, domain, range, hint});
```